pattoo/db/schema/auth.py

```diff
@@ -0,0 +1,71 @@
+"""pattoo ORM Schema for User Authentication"""
+
+# PIP3 imports
+import graphene
+from graphql import GraphQLError
+from flask_graphql_auth import (create_access_token, create_refresh_token,
+                                get_jwt_identity,
+                                mutation_jwt_refresh_token_required)
+
+# pattoo imports
+from pattoo.db.table.user import User as UserTable
+
+
+class AuthMutationInput(graphene.InputObjectType):
+    """Arguments for user authentication"""
+
+    username = graphene.String(required=True, description='Username.')
+    password = graphene.String(required=True, description='Password.')
+
+
+class AuthMutation(graphene.Mutation):
+    """Handles user authentication.
+
+    Handles authentication an distribution of access tokens to allow for
+    users to access server resources
+
+    """
+    access_token = graphene.String()
+    refresh_token = graphene.String()
+
+    class Arguments:
+        Input = AuthMutationInput(required=True)
+
+    def mutate(self, info, Input):
+        username, password = Input['username'], Input['password']
+
+        # Username and password authentication
+        _user = UserTable(username)
+        valid_password = _user.valid_password(password)
+
+        # Verifying that authentication was successful and that the user is
+        # enabled
+        if not(_user.exists and valid_password):
+            err = ('''Authentication Failure: incorrect credentials or user not
+                   enabled!''')
+            raise GraphQLError(err)
```

Comment:
Also use `pattoo_shared.log2die` or `pattoo_shared.log2debug`, that way the message appears in the logs. This won't allow proper troubleshooting.

Reply:
@Dev-Dominic